internal/ingress/controller/template/template.go

```diff
@@ -310,6 +328,12 @@ func buildLocation(input interface{}) string {
 		return fmt.Sprintf(`~* ^%s%s`, path, baseuri)
 	}
 
+	if rewriteAnnotationUsed == true {
+		if path == slash {
+			return path
+		}
+		return fmt.Sprintf(`^~ %s`, path)
```

Comment:
As discussed internally, this is not necessary every time there's rewrite annotation used, rather it is necessary only when the rewrite annotation is used on `/` path. If rewrite annotation is used i.e on `/foo`, then the generated location would like `~* ^/foo\/?(?<baseuri>.*)`, which will not match things like `/.well-known/acme/challenge`.

Given this, do you think we should simplify this further and apply `^~` only when there's an annotation on `/` path?

Reply:
technically, the conflict would occur with any leading portion of `/.well-known/acme/challenge`

eg. `/.well-known`

right?

Reply:
and that expands further to any subpath of a path that does NOT use rewrite... so likely simpler logic this way... ie. just check that rewrite is used instead of looking for conflicting paths